zcash_client_backend/src/data_api.rs

```diff
@@ -20,29 +29,103 @@ use zcash_primitives::{
         components::amount::{Amount, BalanceError, NonNegativeAmount},
         Transaction, TxId,
     },
-    zip32::{AccountId, Scope},
-};
-
-use crate::{
-    address::UnifiedAddress,
-    decrypt::DecryptedOutput,
-    keys::{UnifiedAddressRequest, UnifiedFullViewingKey, UnifiedSpendingKey},
-    proto::service::TreeState,
-    wallet::{Note, NoteId, ReceivedNote, Recipient, WalletTransparentOutput, WalletTx},
-    ShieldedProtocol,
+    zip32::Scope,
 };
+use zip32::DiversifierIndex;
 
 use self::chain::CommitmentTreeRoot;
 use self::scanning::ScanRange;
 
 #[cfg(feature = "transparent-inputs")]
 use zcash_primitives::transaction::components::OutPoint;
 
+use blake2b_simd::{Params as blake2bParams, OUTBYTES as BLAKE2B_OUTBYTES};
+
 pub mod chain;
 pub mod error;
 pub mod scanning;
 pub mod wallet;
 
+/// A hash of a seed used for an HD account.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct HDSeedFingerprint([u8; BLAKE2B_OUTBYTES]);
```

Comment:
Please move this type to `zcash_keys::keys` and make sure to add a CHANGELOG entry there.